promql/parse.go

```diff
@@ -323,9 +326,15 @@ func (p *parser) recover(errp *error) {
 	e := recover()
 	if e != nil {
 		if _, ok := e.(runtime.Error); ok {
-			panic(e)
+			// Print the stack trace but do not inhibit the running application.
+			buf := make([]byte, 64<<10)
+			buf = buf[:runtime.Stack(buf, false)]
+
+			log.Errorf("parser panic: %v\n%s", e, buf)
+			*errp = fmt.Errorf("unexpected error")
```

Comment:
In general Prometheus users are likely trusted by the admins. But in case they are not, exposing runtime issues is probably not a good idea.
Runtime errors are useless without the stack trace anyway.